### ISSUE TITLE
Suggested patch for sets of options

### ISSUE THREAD

**Issue by JohnAtFenestra** [deseretbook/classy_hash#8]
While working with ClassyHash on a project, I encountered a need for something a bit more straightforward for handling a key with a set of allowed options in the form of Symbols. I wanted to be able to write something like:

```
schema = {
  test: Set.new([:hello, :world])
}

source = {
  test: :hello
}
```

After a little research, I figured out that injecting the following into classy_hash.rb solves the problem very easily:

```
  def self.check_one(key, value, constraint, parent_path=nil)
    case constraint
...
    when Set
      self.raise_error(parent_path, key, "is not valid. The value #{value} is not a member of #{constraint.inspect}") unless constraint.include?(value)
```

Example error:

```
/home/john/.rvm/gems/ruby-2.1.5/gems/classy_hash-0.1.5/lib/classy_hash.rb:178:in `raise_error': :hello is not valid. The value 1 is not a member of #<Set: {:one, :two}> (RuntimeError)
```

I'll be using this as a patched version of ClassyHash and wanted to offer it to you as a possible addition.

Best,

John Elrick

**Comment by mike-bourgeous** [comment]
This looks like an interesting and simple way to add enumerations without having to use `CH::G.enum`.  I wouldn't be surprised if your way is faster, too.  I'll definitely consider adding it to ClassyHash when I next work on it.

**Comment by mike-bourgeous** [comment]
@JohnAtFenestra What do you think of #14?  It'll be merged to master for version 0.2.0 of ClassyHash.

**Comment by JohnAtFenestra** [comment]
Sorry I took so long to respond. I forgot to change something on my GitHub settings.
